src/net_processing.cpp

```diff
@@ -1165,23 +1180,27 @@ bool ProcessMessage(CNode *pfrom, std::string strCommand, CDataStream &vRecv, in
             return true;
         }
 
-        // Put the tx on the tx admission queue for processing
-        CTxInputData txd;
-        vRecv >> txd.tx;
+        // Process as many concatenated txns as there may be in this message
+        while (!vRecv.empty())
```

Comment:
shouldn't we do the `while` conditionally to the fact that the peer support or not concatenate txns? 
(maybe it's useless because we know that the sending peer should send us concats txs only if we declare to support it via xversion)

Reply:
I left it this way because "our" peer supports concatentated txns so it doesn't matter whether the remote peer sends us any or not or even whether they support concatentated txns.  However when we send any then we need to know whether the remote peer will support our concat txns or not.